packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/e2e/quanticCategoryFacet.e2e.ts

```diff
@@ -0,0 +1,194 @@
+import {testSearch, testInsight, expect} from './fixture';
+import {
+  useCaseEnum,
+  useCaseTestCases,
+} from '../../../../../../playwright/utils/useCase';
+import facetData from './data';
+
+const fixtures = {
+  search: testSearch,
+  insight: testInsight,
+};
+
+useCaseTestCases.forEach((useCase) => {
+  let test = fixtures[useCase.value];
+  test.describe(`quantic category facet ${useCase.label}`, () => {
+    test.describe('when selecting and deselecting a facet value', () => {
+      test('when selecting a facet value and clicking the all categories button', async ({
+        baseFacet,
+        facet,
+      }) => {
+        const selectedIndex = 0;
+        const {facetId, field, values} = facetData;
+        const expectedFacetData = {
+          facetId,
+          facetField: field,
+          facetValue: values[selectedIndex].value,
+        };
+        const facetSelectUaRequest =
+          baseFacet.waitForFacetSelectUaAnalytics(expectedFacetData);
+        const firstSearchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.clickOnFacetValue(selectedIndex);
+
+        await facetSelectUaRequest;
+        const firstSearchResponse = await firstSearchResponsePromise;
+        const {analytics: analyticsForFacetSelect} =
+          baseFacet.extractDataFromResponse(firstSearchResponse);
+        expect(analyticsForFacetSelect.actionCause).toEqual('facetSelect');
+        if (useCase.value === useCaseEnum.search) {
+          expect(analyticsForFacetSelect.customData).toEqual(
+            expect.objectContaining(expectedFacetData)
+          );
+        }
+
+        const FacetClearAllUaRequest =
+          baseFacet.waitForFacetClearAllUaAnalytics({
+            facetId,
+            facetField: field,
+          });
+        const secondSearchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.allCategoriesButton.click();
+
+        await FacetClearAllUaRequest;
+        const secondSearchResponse = await secondSearchResponsePromise;
+        const {analytics: analyticsForFacetDeselect} =
+          baseFacet.extractDataFromResponse(secondSearchResponse);
+        expect(analyticsForFacetDeselect.actionCause).toEqual('facetClearAll');
```

Comment:
we should add a check or a test to make sure the facet value is set in the URL when clicked